zeebe/broker/src/test/java/io/camunda/zeebe/broker/backup/InMemoryMockBackupStore.java

```diff
@@ -0,0 +1,127 @@
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.broker.backup;
+
+import io.camunda.zeebe.backup.api.Backup;
+import io.camunda.zeebe.backup.api.BackupIdentifier;
+import io.camunda.zeebe.backup.api.BackupIdentifierWildcard;
+import io.camunda.zeebe.backup.api.BackupStatus;
+import io.camunda.zeebe.backup.api.BackupStatusCode;
+import io.camunda.zeebe.backup.api.BackupStore;
+import io.camunda.zeebe.backup.common.BackupStatusImpl;
+import java.io.FileNotFoundException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class InMemoryMockBackupStore implements BackupStore, AutoCloseable {
+
+  private final ConcurrentHashMap<BackupIdentifier, CompletableFuture<Void>> saveFutures =
+      new ConcurrentHashMap<>();
+
+  private final ConcurrentHashMap<BackupIdentifier, BackupStatus> backupStatusMap =
+      new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<BackupIdentifier, Backup> backupMap = new ConcurrentHashMap<>();
+
+  public Set<BackupIdentifier> backupInProgress() {
+    return backupMap.keySet();
+  }
+
+  @Override
+  public CompletableFuture<Void> save(final Backup backup) {
+    backupMap.put(backup.id(), backup);
+    final var fut = new CompletableFuture<Void>();
+    saveFutures.put(backup.id(), fut);
+    return fut;
+  }
+
+  @Override
+  public CompletableFuture<BackupStatus> getStatus(final BackupIdentifier id) {
+    return CompletableFuture.completedFuture(backupStatusMap.get(id));
+  }
+
+  @Override
+  public CompletableFuture<Collection<BackupStatus>> list(final BackupIdentifierWildcard wildcard) {
+    return CompletableFuture.completedFuture(
+        backupStatusMap.entrySet().stream()
+            .filter(e -> wildcard.matches(e.getKey()))
+            .map(Entry::getValue)
+            .toList());
+  }
+
+  @Override
+  public CompletableFuture<Void> delete(final BackupIdentifier id) {
+    backupStatusMap.remove(id);
+    return CompletableFuture.completedFuture(null);
+  }
+
+  @Override
+  public CompletableFuture<Backup> restore(final BackupIdentifier id, final Path targetFolder) {
+    return CompletableFuture.completedFuture(backupMap.get(id));
+  }
+
+  @Override
+  public CompletableFuture<BackupStatusCode> markFailed(
+      final BackupIdentifier id, final String failureReason) {
+    backupStatusMap.computeIfPresent(
+        id,
+        (k, status) ->
+            new BackupStatusImpl(
+                k,
+                status.descriptor(),
+                BackupStatusCode.FAILED,
+                Optional.of(failureReason),
+                status.created(),
+                Optional.of(Instant.now())));
+    return null;
```

Comment:
:question: Does it make sense to return null? If we don't care about this implementation, it's better to throw a `new UnsupportedOperationException("Not yet implemented; implemented it when required")`. It's generally how we introduce new stubs progressively, by only implementing the subset we care about and throwing elsewhere with a clear message, so if someone ever needs this new use case, they can implement it properly.